### PR TITLE
Fix CentOS compilation errors for hip-clang

### PR DESCRIPTION
**Pull request by amdkila**
With the changes in this PR, rocblas can be built using the hip-clang compiler on CentOS. I'll merge it after it passes CI.

### CODE REVIEW

File: clients/benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -143,20 +134,11 @@ if( CMAKE_CXX_COMPILER MATCHES ".*/hcc$" )
...
 endif( )
 
-if( OS_ID_rhel OR OS_ID_centos)
-    # force clang includes to take precedence over devtoolset-7 which we only want for OpenMP
-    set(CMAKE_CXX_FLAGS "-isystem ${CLANG_INCLUDE_DIR} -isystem ${XXX_OPENMP_INCLUDE_DIR} ${CMAKE_CXX_FLAGS}")
-endif()
-
-# include order workaround to force /opt/rocm/include later in order to ignore installed rocblas
-set(CMAKE_CXX_FLAGS "-isystem /opt/rocm/include ${CMAKE_CXX_FLAGS}")
-
```
Comment (TorreZuk) [inline clients/benchmarks/CMakeLists.txt]: This we will likely still want to allow ignoring an installed rocblas when building

File: clients/benchmarks/CMakeLists.txt [hunk 2]
```diff
@@ -143,20 +134,11 @@ if( CMAKE_CXX_COMPILER MATCHES ".*/hcc$" )
...
 elseif( CMAKE_COMPILER_IS_GNUCXX OR CXX_VERSION_STRING MATCHES "clang")
   # GCC or hip-clang needs specific flags to turn on f16c intrinsics
   target_compile_options( rocblas-bench PRIVATE -mf16c )
 endif( )
 
-if( OS_ID_rhel OR OS_ID_centos)
-    # force clang includes to take precedence over devtoolset-7 which we only want for OpenMP
-    set(CMAKE_CXX_FLAGS "-isystem ${CLANG_INCLUDE_DIR} -isystem ${XXX_OPENMP_INCLUDE_DIR} ${CMAKE_CXX_FLAGS}")
-endif()
```
Comment (TorreZuk) [inline clients/benchmarks/CMakeLists.txt]: This is a separate issue which you are testing to see if they removed the conflicting files.  Good to test and resolve/remove it as it was intended as a temporary workaround.

File: clients/benchmarks/CMakeLists.txt [hunk 3]
```diff
@@ -66,60 +66,51 @@ target_include_directories( rocblas-bench
 set( BLIS_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/build/deps/blis/include/blis )
 set( BLIS_LIBRARY ${CMAKE_SOURCE_DIR}/build/deps/blis/lib/libblis.so )
 
-
-if( OS_ID_rhel OR OS_ID_centos OR OS_ID_sles)
+if( OS_ID_rhel OR OS_ID_sles OR OS_ID_centos)
     if( OS_ID_rhel OR OS_ID_centos)
         # defer OpenMP include as search order must come after clang
-        set( XXX_OPENMP_INCLUDE_DIR /opt/rh/devtoolset-7/root/usr/lib/gcc/x86_64-redhat-linux/7/include )
```
Comment (TorreZuk) [inline clients/benchmarks/CMakeLists.txt]: XXX was workaround to reorder includes, see later comment.  Good to remove now if it works with latest compiler and rocm install.

File: clients/gtest/CMakeLists.txt [hunk 4]
```diff
@@ -4,7 +4,11 @@
 
 # For debugging, uncomment this
 # set( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -g -O0" )
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
+if(EXISTS /etc/redhat-release)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp=libgomp -pthread -stdlib=libstdc++ -std=c++11")
```
Comment (TorreZuk) [inline clients/gtest/CMakeLists.txt]: You should test without the c++11, and even the -stdlib could be inside another clang only guard if you wanted.

File: clients/benchmarks/CMakeLists.txt (hunk 1 above)
Comment (amdkila) [reply clients/benchmarks/CMakeLists.txt]: The -isystem arguments break the build on CentOS, causing those C++ STL compilation errors. I could hide them for CentOS

**Pull request merged**